dali/pipeline/data/tensor_vector.h

```diff
@@ -223,21 +264,97 @@ class DLL_PUBLIC TensorVector {
   DLL_PUBLIC void UnsafeCopySample(int sample_idx, const TensorVector<Backend> &src,
                                    int src_sample_idx, AccessOrder order = {});
 
+  DLL_PUBLIC void UnsafeCopySample(int sample_idx, const Tensor<Backend> &src,
+                                   AccessOrder order = {});
+  /** @} */
+
+
+  /**
+   * @name Type access
+   * @{
+   */
+  /**
+   * @brief Set the type of the current batch. The type needs to be set before calling
+   * the Resize(const TensorListShape<> &) function. It cannot be used to change the type after
+   * allocation happened.
+   *
+   * Resize(const TensorListShape<> &, DALIDataType) can be used without prior set_type call or to
+   * request a different type after allocation.
+   */
+  void set_type(DALIDataType new_type);
+
+  template <typename T>
+  void set_type() {
+    set_type(TypeTable::GetTypeId<T>());
+  }
+
+  /**
+   * @brief Get the type of samples in the batch.
+   *
+   * @note Using DALIDataType is recommended over accessing type_info().
```

Comment:
Nitpick: This is nice remark but I think the better place for it is type_info docstring. It is a bit as if we put deprecation note in the new functionality telling "good you are reading this because that another funcionality is deprecated".

Reply:
done